Python/optimizer_analysis.c

```diff
@@ -83,6 +1101,93 @@ remove_unneeded_uops(_PyUOpInstruction *buffer, int buffer_size)
     }
 }
 
+static inline bool
+op_is_zappable(int opcode)
+{
+    switch(opcode) {
+        case _SET_IP:
```

Comment:
I think this is missing `_NOP`. It seems a fine opcode to back over in the peepholer. :-)

Reply:
Interestingly, after I do this, I believe we won't need to run the peepholer more than once. (I put in `assert(peephole_attempts == 0 || done);` after the call and it didn't fail a thing.) SO that would simplify things a bit. 

I also have a more radical idea: we can move the two things that the peepholer currently does (`_ to hand-written cases in `uop_abstract_interpret_single_inst()`. Then we don't need the peepholer at all any more.

Reply:
Right now actually the peepholer only needs to run once. I just let it run a few times just to be sure.

My intuition why it can eliminate all `_SHRINK_STACK` on the first run is that since we are operating on a bytecode stack machine IR, which is operating forwards.

Reply:
But see my [counter-example](https://github.com/faster-cpython/ideas/issues/648#issuecomment-1918169349) of a walrus: `A + (X := B*C)`.

Either way it seems we both agree that `_SHRINK_STACK` is a temporary crutch.